### ISSUE TITLE
Add generalizable sidenav/content MDX layout

### ISSUE THREAD

**Issue by stephenkao** [CDCgov/prime-reportstream#8943]
## Problem statement

With MDX support, we'll be able to generalize our documentation pages through reusable layouts rather than the existing IATemplate logic.

## What you need to know

- This is intended to replace the existing [IATemplate logic](https://github.com/CDCgov/prime-reportstream/blob/master/frontend-react/src/components/Content/Templates/IATemplate.tsx#L24-L63) in the future

## Acceptance criteria

- [ ] There are MDX templates for sidebar navigation and page content
    - [ ] Nice-to-have: The templates are generalizable for any content page

## Follow-ups
- Convert existing Markdown pages to MDX
    - [#8952](https://app.zenhub.com/workspaces/web-experience-607d9d5e68b95200150fec37/issues/gh/cdcgov/prime-reportstream/8952)
    - [#8953](https://app.zenhub.com/workspaces/web-experience-607d9d5e68b95200150fec37/issues/gh/cdcgov/prime-reportstream/8953)

**Comment by chris-kuryak** [comment]
Waiting on one more review before PR merge.
